src/check/context/clss/concrete/variant.rs

```diff
@@ -32,10 +32,10 @@ impl From<&ClassVariant> for Name {
     fn from(class_variant: &ClassVariant) -> Self {
         let name_variant = match &class_variant {
             ClassVariant::Direct(class) if class.len() == 1 => {
-                class.iter().next().unwrap().name.variant.clone()
+                NameVariant::Single(class.iter().next().unwrap().name.clone())
             }
             ClassVariant::Direct(classes) => {
```

Comment:
Not strictly speaking part of this PR, but why do we even differentiate between these two cases?